server/schedulers/hot_region.go

```diff
@@ -606,184 +660,258 @@ func (bs *balanceSolver) getDstCandidateIDs() map[uint64]struct{} {
 			filter.NewHealthFilter(bs.sche.GetName()),
 		}
 
-		candidates = bs.cluster.GetFollowerStores(bs.region)
+		candidates = bs.cluster.GetFollowerStores(bs.cur.region)
 
 	default:
 		return nil
 	}
 
-	ret := make(map[uint64]struct{}, len(candidates))
+	ret := make(map[uint64]*storeLoadDetail, len(candidates))
 	for _, store := range candidates {
 		if !filter.Target(bs.cluster, store, filters) {
-			ret[store.GetID()] = struct{}{}
+			ret[store.GetID()] = bs.stLoadDetail[store.GetID()]
 		}
 	}
 	return ret
 }
 
-func (bs *balanceSolver) selectDstStoreID(candidateIDs map[uint64]struct{}) uint64 {
-	candidateLoadDetail := make(map[uint64]*storeLoadDetail, len(candidateIDs))
-	for id := range candidateIDs {
-		candidateLoadDetail[id] = bs.stLoadDetail[id]
-	}
-	switch bs.opTy {
-	case movePeer:
-		return selectDstStoreByByteRate(candidateLoadDetail, bs.srcPeerStat.GetByteRate(), bs.stLoadDetail[bs.srcStoreID])
-	case transferLeader:
-		if bs.rwTy == write {
-			return selectDstStoreByCount(candidateLoadDetail, bs.srcPeerStat.GetByteRate(), bs.stLoadDetail[bs.srcStoreID])
+// calcProgressiveRank checks `bs.cur` is a progressive solution.
+func (bs *balanceSolver) calcProgressiveRank() int64 {
```

Comment:
add some comments about how or/and why ranks are determined?

Reply:
I will add comments about meaning of the "rank". As the implementation of this function is desired to change in the future,  I think "how or why ranks are determined" is too  detailed as function comment. I will add "how or why"  comment in the implementation.

Reply:
Comments added.